src/main/java/net/neoforged/neoforge/event/EventHooks.java

```diff
@@ -303,13 +303,13 @@ public static boolean checkSpawnPositionSpawner(Mob mob, ServerLevelAccessor lev
     @SuppressWarnings("deprecation") // Call to deprecated Mob#finalizeSpawn is expected.
     public static SpawnGroupData finalizeMobSpawn(Mob mob, ServerLevelAccessor level, DifficultyInstance difficulty, MobSpawnType spawnType, @Nullable SpawnGroupData spawnData) {
         var event = new FinalizeSpawnEvent(mob, level, mob.getX(), mob.getY(), mob.getZ(), difficulty, spawnType, spawnData, null);
-        boolean cancel = NeoForge.EVENT_BUS.post(event).isCanceled();
+        NeoForge.EVENT_BUS.post(event);
 
-        if (!cancel) {
-            mob.finalizeSpawn(level, event.getDifficulty(), event.getSpawnType(), event.getSpawnData());
+        if (!event.isCanceled()) {
+            return mob.finalizeSpawn(level, event.getDifficulty(), event.getSpawnType(), event.getSpawnData());
         }
 
-        return cancel ? null : event.getSpawnData();
+        return null;
```

Comment:
The new implementation no longer allows mods to force custom `SpawnGroupData` from their event handler to be used.

Reply:
It gets used by the `Mob#finalizeSpawn` call, which is then responsible for propagating it forward.  Typically those implementations just return the passed spawn group data.